### ISSUE TITLE
Add-Ons Manager: Allow Users to Undo Add-on Installation before Restarting NVDA

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10025]
This is essentially the same as #10016, but concerning installing rather than uninstalling.

When installing an add-on via the Add-ons Manager, a user is first asked to confirm that they're happy to install the add-on; if choosing "Yes", the Add-ons Manager then prompts that NVDA needs to be restarted before installation can complete. If a user chooses to install an add-on but then changes their mind before initiating the restart of NVDA (for example they realise that the add-on is Malicious), we could add an "Undo" button to the Add-ons Manager to ensure that the chosen add-on would not be installed and activated upon the next NVDA restart.

**Comment by seanbudd** [comment]
Closing as can't fix.
At this state, install tasks have been run by NVDA and the add-on, the installation process has started. The installation needs to finish on restart before a safe uninstall can occur. Unfortunately these tasks cannot be delayed until restart as  discussed in #14430. The install operation must be atomic between restart due to add-ons relying on install tasks.
See also #15578, which is a case that can be cancelled.
